libmscore/segment.cpp

```diff
@@ -2183,22 +2183,32 @@ qreal Segment::minHorizontalDistance(Segment* ns, bool systemHeaderGap) const
                   w = ww + score()->styleP(Sid::clefLeftMargin);
                   }
             else {
-                  bool isGap = false;
+                  bool possibleGap = false;
+                  bool notGap = false;
+                  bool isAdjacent = true;
                   for (int i = 0; i < score()->nstaves() * VOICES; i++) {
                         Element* el = element(i);
                         if (!el)
                               continue;
                         if (el->isRest() && toRest(el)->isGap())
-                              isGap = true;
+                              possibleGap = true;
                         else {
-                              isGap = false;
-                              break;
+                              notGap = true;
+                              if (ns && ns->element(i)) {
+                                    isAdjacent = true;
+                                    break;
+                                    }
                               }
                         }
-                  if (isGap)
+                  if (possibleGap && !notGap)
                         return 0.0;
-                  // minimum distance between notes is one note head width
-                  w = qMax(w, score()->noteHeadWidth()) + score()->styleP(Sid::minNoteDistance);
+                  // minimum distance between notes was one note head width
+                  // this should not be necessary given we are calculating actual segment distances
+                  // also, minNoteDistance is only needed in cases where there are adjacent notes
+                  // in the same voice
+                  //w = qMax(w, score()->noteHeadWidth()) + score()->styleP(Sid::minNoteDistance);
+                  if (isAdjacent)
+                        w = w + score()->styleP(Sid::minNoteDistance);
```

Comment:
Where exactly would we need to check for an older version and enable compatibility?

Reply:
I have no idea at this point.  I don't know that there is a clearly laid out plan for this, but there should be, as it's likely to come up any number of other places we might want to improve default spacing for new scores without adversely affecting older ones.  We waited *way* too long to sort that out for 3.6 with the new style defaults, hence the cascade of bugs during the last couple of weeks before and after release.

Reply:
I'd really like to include it in #9000, but am not sure about its implications and whether and how to make sure to not screw up pre 3.6.3 layouts